industry_benchmarks/utils/fix_networks.py

```diff
@@ -0,0 +1,603 @@
+
+    Returns
+    -------
+    alchemical_network : gufe.AlchemicalNetwork
+      A deduplicated alchemical network.
+    """
+    transform_list = []
+
+    for t_name, t_list in transforms_dict.items():
+        if len(t_list) != 3:
+            # TODO: we can turn this into a warning message if it's too painful
+            errmsg = (
+                f"Too few transformations found for {t_name} "
+                "this indicates a partially completed set of results. "
+                "Please ensure that your input network is finished and "
+                "any reproducible partial failures have been removed."
+            )
+            raise ValueError(errmsg)
+        if not all(a == t_list[0] for a in t_list):
+            errmsg = (
+                f"Transfromations for {t_name} do not match "
+                "this may indicate some kind of corruption of the alchemical "
+                "network. Please contact the OpenFE team."
+            )
+            raise ValueError(errmsg)
+
+        transform_list.append(t_list[0])
+
+    return AlchemicalNetwork(transform_list)
+
+
+def parse_results(
+    result_files: list[str],
+    input_ligand_network: LigandNetwork,
+) -> AlchemicalNetwork:
+    """
+    Create an AlchemicalNetwork from a set of input JSON files.
+    """
+    # All the transforms, triplicated
+    all_transforms_dict = {}
+
+    # Fail if we have no inputs
+    if len(result_files) == 0:
+        errmsg = "No results .json files were provided"
+        raise ValueError(errmsg)
+
+    for resfile in result_files:
+        ru = _get_check_results_json(resfile)
+
+        if ru is None:
+            continue
+
+        # TODO: fix this, for now this will just fail
+        transform, phase = get_transformation(ru)
+
+        if transform is None:
+            # We delete the inputs data
+            transform, phase = get_transformation_alternate(
+                ru, input_ligand_network
+            )
+
+        if transform.name in all_transforms_dict:
+            all_transforms_dict[transform.name].append(transform)
+        else:
+            all_transforms_dict[transform.name] = [transform]
+
+    alchemical_network = _check_and_deduplicate_transforms(all_transforms_dict)
+
+    return alchemical_network
+
+
+def alchemical_network_to_ligand_network(alchemical_network) -> LigandNetwork:
+    """
+    Create a LigandNetwork from an AlchemicalNetwork
+    """
+    edges = []
+    for e in alchemical_network.edges:
+        edges.append(e.mapping)
+
+    network = LigandNetwork(edges=set(edges))
+    return network
+
+
+def decompose_disconnected_ligand_network(
+    ligand_network: LigandNetwork,
+) -> list[LigandNetwork]:
+    """
+    Get a list of connected networks from an input
+    disconnected network
+    """
+    # build alchemical sub-networks
+    g = ligand_network.graph
+    all_edges = list(ligand_network.edges)
+    sub_network_nodes = list(nx.connected_components(g))
```

Comment:
This is failing because it's a multidigraph I think - we can just create a digraph from g and it should work

Reply:
Now first creating an undirected graph from g